### ISSUE TITLE
if-else does not work properly (regression)

### ISSUE THREAD

**Issue by skylerberg** [redox-os/ion#158]
Running this script:
```
if a == a
  echo true a == a
  if b != b
    echo true b != b
  else
    echo false b != b
    if 3 > 2
      echo true 3 > 2
    else
      echo false 3 > 2
    end
  end
else
  echo false a == a
end
```
Results in:
```
true a == a
false b != b
No comparison operator
Syntax error: else found with no previous if
false a == a
Syntax error: end found outside of a block
```

This is really wrong. 

I think this answer is to move this into the grammar, but it is not going to be easy: https://github.com/redox-os/ion/issues/119. Also, https://github.com/redox-os/ion/issues/66 could have prevented this regression.

**Comment by mmstick** [comment]
I'm not at all familiar with the grammar, but is there some documentation to explain how the grammar works?

**Comment by nounoursheureux** [comment]
My guess is that the parser makes the confusion between `<` and `>` used for comparing two values and `<` and `>` used for piping (the `==` and `!=` comparison operators work fine)

**Comment by skylerberg** [comment]
@mmstick Unfortunately, the grammar is a real mess and it is hard to follow. It is in peg.rs and the best documentation for understanding how it works is to read up on Parsing Expression Grammars and rust-peg (a dependencies for Ion).

@nounoursheureux Ah, that is a good guess.

**Comment by nounoursheureux** [comment]
Maybe I'm missing something, but I the only way that I see to fix this issue is to use other comparison operators

**Comment by mmstick** [comment]
Bash at least just uses a backslash for escaping the `<`, while conditionals are performed using the test command which may or may not be built in.

**Comment by skylerberg** [comment]
We might be able to fix this with https://github.com/redox-os/ion/issues/119. That way the grammar would expect an operator in an if statement and would treat it as an operator.
